src/service/data_products/genome_attributes.py

```diff
@@ -140,11 +158,24 @@ async def get_ranks(
     # could get the doc count, but that'll be slower since all docs have to be counted vs. just
     # getting LIMIT docs. YAGNI for now
     cur = await store.aql().execute(aql, bind_vars=bind_vars)
-    ret = []
+    # Sort everything since we can't necessarily rely on arango, the client, or the loader
+    # to have the same insertion order for the dicts
+    # If we want a specific order the loader should stick a keys doc or something into arango
+    # and we order by that
+    data = []
+    d = None
     async for d in cur:
-        # not clear there's any benefit to creating a bunch of TaxaCount objects here
-        ret.append(remove_collection_keys(remove_arango_keys(d)))
-    if ret and sort_on not in ret[0]: 
-        raise errors.IllegalParameterError(
-            f"No such field for collection {collection_id} load version {load_ver}: {sort_on}")
-    return {_FLD_SKIP: skip, _FLD_LIMIT: limit, "data": ret}
+        if output_table:
+            data.append([d[k] for k in sorted(_remove_keys(d))])
+        else:
+            data.append({k: d[k] for k in sorted(_remove_keys(d))})
+    fields = []
+    if d:
+        if sort_on not in d: 
+            raise errors.IllegalParameterError(
+                f"No such field for collection {collection_id} load version {load_ver}: {sort_on}")
+        fields = [{"name": k} for k in sorted(d)]
```

Comment:
why not just assign `fields = sorted(d)`?

Reply:
I think this is to allow adding other field properties (i.e. sort/filter type, human-readable name) in the future without it being a breaking change (correct me if I'm wrong @MrCreosote)

Reply:
right

Reply:
👍